src/Conversion/ONNXToTOSA/Tensor/Constant.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Constant.cpp - Const Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX const operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXConstOpLoweringToTOSA : public OpConversionPattern<ONNXConstantOp> {
+public:
+  using OpConversionPattern<ONNXConstantOp>::OpConversionPattern;
+  using OpAdaptor = typename ONNXConstantOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXConstantOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    ::mlir::Attribute valueAttr = *adaptor.value();
+    mlir::Type resultType = getTypeConverter()->convertType(op.getResult().getType());
```

Comment:
As discussed in the call, assuming `value` is present is too strong.

The crash on `resultType` seems reasonable to me as there is no reason to expect the conversion to fail if the arguments passed type conversion. I do not think this needs additional checking (other than seeing if the type is needed, or if it is guaranteed to be the same as that of `valueAttr`).